components/Forms/ComponentFactory.tsx

```diff
@@ -209,24 +221,33 @@ export const ComponentFactory: React.VFC<FactoryProps> = observer(
                   form.clearForm()
                 }}
               >
-                Clear
+                {tsln.clear}
               </button>
               <button
                 type="submit"
                 role="button"
                 className="btn btn-primary mt-4 md:mt-0 col-span-2 md:col-span-1 disabled:cursor-not-allowed disabled:bg-[#949494] disabled:border-0"
                 onClick={async () => {
-                  if (!form.validateAgainstEmptyFields() && !form.hasErrors) {
-                    selectedTabIndex(1)
+                  if (
+                    !form.validateAgainstEmptyFields(router.locale) &&
+                    !form.hasErrors
+                  ) {
+                    const language = document.querySelector(
+                      '#_language'
+                    ) as HTMLInputElement
+                    root.setCurrentLang(
+                      language.value == 'EN' ? Language.EN : Language.FR
```

Comment:
isn't this redundant? can't it just be `language.value`? as it is always either EN or FR

Reply:
This was me juking the mst type. `string !== Language` but I can just use `as` instead.